File: FlashCards.Business/DTOs/CreateCardDTO.cs
using FlashCards.Core.Enums;

namespace FlashCards.Business.DTOs
{
    public class CreateCardDTO
    {
        public Guid? Id { get; set; }
        public string FrontText { get; set; }
        public string BackText { get; set; }
        public Guid DeckId { get; set; }
        public CardLanguage Language { get; set; }
        public GrammaticalGender Gender { get; set; }
        public string? Plural { get; set; }
        public string? Pronunciation { get; set; }
    }
}

File: german-cards-pwa/src/app/models/create-card-dto.ts
import { CardLanguage } from './card-language';
import { GrammaticalGender } from './grammatical-gender';

export interface CreateCardDto {
  id?: string;
  frontText: string;
  backText: string;
  deckId: string;
  language: CardLanguage;
  gender: GrammaticalGender;
  plural?: string | null;
  pronunciation?: string | null;
}
